### ISSUE TITLE
Editor functionality of ClipTransition scriptable object assets are broken

### ISSUE THREAD

**Issue by mode80** [KybernetikGames/animancer#18]
When making a new ClipTransition with the Create->Clip Transition menu, the resulting scriptable object assets is missing its link to the source script. (image attached) The asset itself seems work as expected at runtime. However some key editor functionality is broken as a result: The default editor for scriptable objects shows up rather than your custom inspector editor. Also when duplicating (cmd+D) such an asset, the newly created asset is missing all fields and is nonfunctional. 

I tried renaming the file to ClipTransition.cs and everything worked as expected...

Ag..just realized you maybe fixed this already in Beta 6. If so feel free to close!

<img width="399" alt="Screen Shot 2020-01-15 at 10 24 55 AM" src="https://user-images.githubusercontent.com/656321/72456145-4c61ad00-3781-11ea-9e34-16f4014528bd.png">

**Comment by mode80** [comment]
FYI Anyone upgrading from Beta5 will find their previously crated ClipTransition assets borked as a result of this fix. Not sure there's any workaround short of recreating them all, but might be worth a bullet in the change log.

**Comment by KybernetikGames** [comment]
There's nothing in the change log because I haven't changed it recently. In fact, according to my commit history I haven't touched that file at all since it was first created in Beta 4.

The only explanation I can think of would be if you had compile errors, but even then I wouldn't expect it to cause something like this.

**Comment by KybernetikGames** [comment]
Actually, a more likely possibility I just thought of is if you didn't delete your previous version of Animancer before importing Beta 5.

There used to only be one transition asset called `AnimancerTransition` which did what `ClipTransition` does now, so when I created all the new ones I gave the `ClipTransition` script the asset GUID from `AnimancerTransition` so that any existing ones should automatically change to the new class without losing their data.

But if you import the new version of Animancer without deleting the old one first, Unity's stupid importer will see that a different file exists with the same GUID and will update the contents of the file but will not rename it (or move it to the new location). So you end up with the script file still called `AnimancerTransition.cs` and its class called `ClipTransition` which Unity doesn't support for `MonoBehaviour`s and `ScriptableObject`s. I've been using Unity for about 10 years now and I still have no idea why that restriction even exists or why they thought it was a good idea for the package importer to work like that.
